src/app/features/issue/providers/caldav/caldav-common-interfaces.service.ts

```diff
@@ -0,0 +1,153 @@
+import {Injectable} from '@angular/core';
+import {Observable, of} from 'rxjs';
+import {Task} from 'src/app/features/tasks/task.model';
+import {IssueServiceInterface} from '../../issue-service-interface';
+import {SearchResultItem} from '../../issue.model';
+import {CaldavIssue, CaldavIssueReduced} from './caldav-issue/caldav-issue.model';
+import {CaldavClientService} from './caldav-client.service';
+import {CaldavCfg} from './caldav.model';
+import {catchError, concatMap, first, switchMap} from 'rxjs/operators';
+import {ProjectService} from '../../../project/project.service';
+import {SnackService} from '../../../../core/snack/snack.service';
+import {truncate} from '../../../../util/truncate';
+import {T} from '../../../../t.const';
+
+
+@Injectable({
+              providedIn: 'root',
+            })
+export class CaldavCommonInterfacesService implements IssueServiceInterface {
+  constructor(
+    private readonly _projectService: ProjectService,
+    private readonly _caldavClientService: CaldavClientService,
+    private readonly _snackService: SnackService,
+  ) {
+  }
+
+  getAddTaskData(issueData: CaldavIssueReduced): { title: string; additionalFields: Partial<Task> } {
+    return {additionalFields: {issueLastUpdated: issueData.last_modified}, title: issueData.summary};
+  }
+
+  getById$(id: string | number, projectId: string): Observable<CaldavIssue> {
+    return this._getCfgOnce$(projectId).pipe(
+      concatMap(caldavCfg => this._caldavClientService.getById$(id, caldavCfg))
+    );
+  }
+
+  issueLink$(issueId: string | number, projectId: string): Observable<string> {
```

Comment:
I have never worked with cal dav before. But is there no UI within nextcloud for accessing those? If not we might want to make issueLink$ nullable and only display the option to go to an issue if the link is available.

Reply:
There is a UI within NC - but NC is not the only caldav server. I used davical for a long time for example.